### PR TITLE
libblkid: (hfsplus) reduce false positive

### PR DESCRIPTION
**Pull request by karelzak**
Fixes: https://github.com/util-linux/util-linux/issues/2692

### CODE REVIEW

File: libblkid/src/superblocks/hfs.c [hunk 1]
```diff
@@ -238,17 +242,23 @@ static int probe_hfsplus(blkid_probe pr, const struct blkid_idmag *mag)
 	    (memcmp(hfsplus->signature, "HX", 2) != 0))
 		return 1;
 
-	hfs_set_uuid(pr, hfsplus->finder_info.id, sizeof(hfsplus->finder_info.id));
-
+	/* Block size must be at least as large as a sector and a multiple of 2. */
```
Comment (t-8ch) [inline libblkid/src/superblocks/hfs.c:245]: I didn't find docs for HFS that describe this field in detail, but other FS normally use power-of-two, not multiple-of-two.

File: libblkid/src/superblocks/hfs.c [hunk 2]
```diff
@@ -238,17 +242,23 @@ static int probe_hfsplus(blkid_probe pr, const struct blkid_idmag *mag)
...
 	blocksize = be32_to_cpu(hfsplus->blocksize);
-	if (blocksize < HFSPLUS_SECTOR_SIZE)
+	if (blocksize < HFSPLUS_SECTOR_SIZE || ((blocksize - 1) & blocksize))
```
Comment (t-8ch) [inline libblkid/src/superblocks/hfs.c:247]: `blocksize % 2` would be easier to understand.

Comment (karelzak) [reply libblkid/src/superblocks/hfs.c:247]: Copy & past from kernel: https://github.com/torvalds/linux/blob/master/fs/hfsplus/wrapper.c#L234, the comment needs fix.

Comment (t-8ch) [reply libblkid/src/superblocks/hfs.c:247]: `is_power_of_2()` from `c.h` would be clearer then.

I can send a kernel patch if you prefer.

**Pull request merged**